src/components/board.tsx

```diff
@@ -13,19 +13,42 @@ type Props = {
 export const Board = (props: Props) => {
   const squaresMap = useContext(SquareMapContext);
   const [isClickEnabled, setClickEnabled] = useState(true);
+  const [change, setChange] = useState(0);
   const [getFirstSelectedSquare, setFirstSelectedSquare] =
-    useState<SelectedSquare | null>(null);
+    useState<NullableSelectedSquare>(null);
 
   const [getSecondSelectedSquare, setSecondSelectedSquare] =
-    useState<SelectedSquare | null>(null);
+      useState<NullableSelectedSquare>(null);
 
+  useEffect(() => {
```

Comment:
Maybe we can separate this logic to another hook file similar to how matchChecker.ts is?

Reply:
done